### PR TITLE
doc: fix broken relative md links

### PR DESCRIPTION
**Pull request by willcl-ark**
These relative links in our documentation are broken, fix them.

### CODE REVIEW

**Review by maflcko — APPROVED** [review]
lgtm

File: doc/design/libraries.md [hunk 1]
```diff
@@ -95,7 +95,7 @@ class bitcoin-qt,bitcoind,bitcoin-cli,bitcoin-wallet bold
 
 - The only thing that should depend on *libbitcoin_kernel* internally should be *libbitcoin_node*. GUI and wallet libraries *libbitcoinqt* and *libbitcoin_wallet* in particular should not depend on *libbitcoin_kernel* and the unneeded functionality it would pull in, like block validation. To the extent that GUI and wallet code need scripting and signing functionality, they should be get able it from *libbitcoin_consensus*, *libbitcoin_common*, and *libbitcoin_util*, instead of *libbitcoin_kernel*.
 
-- GUI, node, and wallet code internal implementations should all be independent of each other, and the *libbitcoinqt*, *libbitcoin_node*, *libbitcoin_wallet* libraries should never reference each other's symbols. They should only call each other through [`src/interfaces/`](`../../src/interfaces/`) abstract interfaces.
+- GUI, node, and wallet code internal implementations should all be independent of each other, and the *libbitcoinqt*, *libbitcoin_node*, *libbitcoin_wallet* libraries should never reference each other's symbols. They should only call each other through [`src/interfaces/`](../../src/interfaces/) abstract interfaces.
```
[the remark this answers is not in the document]
Comment (willcl-ark) [reply doc/design/libraries.md:98]: I wanted to but those links a few lines [above](https://github.com/bitcoin/bitcoin/pull/30025/files/8b3e86285f55d484e71a821a2c4538b2f010be59#diff-b5e0af5339b28c1c6c155d36f0157c62232fd994410334fb09bd7511fef12074R85) were relative so I just thought I'd leave these matching 😢 

Will do it now with the other suggestion.

File: doc/developer-notes.md [hunk 2]
```diff
@@ -1335,8 +1335,8 @@ Release notes should be written for any PR that:
 
 Release notes should be added to a PR-specific release note file at
 `/doc/release-notes-<PR number>.md` to avoid conflicts between multiple PRs.
-All `release-notes*` files are merged into a single
-[/doc/release-notes.md](/doc/release-notes.md) file prior to the release.
+All `release-notes*` files are merged into a single `release-notes-<version>.md` file prior to the release, e.g.:
+[/doc/release-notes-27.0.md](release-notes/release-notes-27.0.md).
```
[the remark this answers is not in the document]
Comment (willcl-ark) [reply doc/developer-notes.md]: Done in 8e394d1d3b6ead130515222f5b34d509fff200a8

**Review by ismaelsadeeq — COMMENTED** [review]
utACK 8e394d1d3b6ead130515222f5b34d509fff200a8

**Comment by ryanofsky** [comment]
Code review 8e394d1d3b6ead130515222f5b34d509fff200a8

I don't think it is good to change these to absolute links. When I open these files in emacs I can click the relative links and see the load the target files. If I change the links to be absolute, it tries to open non-existent paths on my root file system.

I imagine emacs is not the only tool which behaves this way. Is it necessary to make links absolute for broken link detection to work, or some other motivation to make them absolute?

**Comment by willcl-ark** [comment]
The `mlc` tool seems fine with relative or absolute, so long as they resolve.

`nvim` seems to jump using relative or absolute links.

I did check to see if one or the other was more markdown-compliant, but it seems they purposefully avoided specifying. Seems like here we might prefer relative so we cater to emacs users, all other things being equal?

**Comment by ryanofsky** [comment]
Oh, ok. I wasn't aware absolute links were already being used. I would slightly prefer links in the two files I wrote (libraries.md and multiprocess.md) to be relative so they work with emacs and so links within the files use a consistent style. But since I am the one responsible for a lot of these broken links, don't weigh my opinion too heavily :wink:

**Comment by willcl-ark** [comment]
> I would slightly prefer links in the two files I wrote (libraries.md and multiprocess.md) to be relative so they work with emacs and so links within the files use a consistent style

A rough search of *.md files shows me that we have 87 absolute links and 72 relative:

<details>
<summary>Details</summary>

```fish
will@ubuntu in ~/src/bitcoin on  fix-links [$!?] : 🐍 (bitcoin)
₿ rg "\]\(\/" */**.md | wc -l
87

will@ubuntu in ~/src/bitcoin on  fix-links [$!?] : 🐍 (bitcoin)
₿ rg "\]\(\.\." */**.md | wc -l
72
```

</details>

...but it seems very reasonable to not break the documents you wrote/lead in maintaining, so I reverted those to relative in 4b9f49da2b120e81516ddc3dc577d7a2e58e02d3.

4b9f49da2b120e81516ddc3dc577d7a2e58e02d3 also fixes a broken external link to flake8 (it seems they deleted the gitlab repo entirely and moved to GitHub).

> Would it be easy to add mlc (or something like it) to the lint CI task? (Just as an idea for a follow-up, not for here)

Yes it should be; I already had a second commit in this branch to run this as a GHA but dropped it (as I thought these are quite unlikely to break regularly).

I could open a PR to have this GHA run [scheduled weekly](https://docs.github.com/en/actions/using-workflows/events-that-trigger-workflows#schedule) (or something like that) if there's interest.

> if mlc (or similar) is added in the future and checks against this, relative paths should be fine (and maybe preferred, if that helps with emacs)?

Yes agree, relative seems better if they are being checked.

File: test/lint/README.md [hunk 3]
```diff
@@ -30,13 +30,13 @@ Then you can use:
...
 | Lint test | Dependency |
 |-----------|:----------:|
-| [`lint-python.py`](lint/lint-python.py) | [flake8](https://gitlab.com/pycqa/flake8)
-| [`lint-python.py`](lint/lint-python.py) | [lief](https://github.com/lief-project/LIEF)
-| [`lint-python.py`](lint/lint-python.py) | [mypy](https://github.com/python/mypy)
-| [`lint-python.py`](lint/lint-python.py) | [pyzmq](https://github.com/zeromq/pyzmq)
-| [`lint-python-dead-code.py`](lint/lint-python-dead-code.py) | [vulture](https://github.com/jendrikseipp/vulture)
-| [`lint-shell.py`](lint/lint-shell.py) | [ShellCheck](https://github.com/koalaman/shellcheck)
-| [`lint-spelling.py`](lint/lint-spelling.py) | [codespell](https://github.com/codespell-project/codespell)
+| [`lint-python.py`](/test/lint/lint-python.py) | [flake8](https://github.com/PyCQA/flake8)
```
Comment (ryanofsky) [inline test/lint/README.md:33]: In commit "doc: fix broken relative md links" (4b9f49da2b120e81516ddc3dc577d7a2e58e02d3)

This is also updating the flake8 project url, not just changing links to be absolute, so could mention this in the commit message

**Review by ryanofsky — APPROVED** [review]
Code review ACK 4b9f49da2b120e81516ddc3dc577d7a2e58e02d3. Thanks for the updates!

**Review by ismaelsadeeq — COMMENTED** [review]
Re ACK 4b9f49da2b120e81516ddc3dc577d7a2e58e02d3

**Pull request merged**